### DISCUSSION TITLE
'Khi Chess' Variant

### DISCUSSION THREAD

**Discussion by SocialSophia** [gbtami/pychess-variants#1077] [Ideas]
You can adjust the variant name if you'd prefer. It's based off S-Chess with the Musketeer style pieces that everyone loves using the archbishop and chancellor replacing the hawk and elephant respectively.

The pieces: https://www.chessvariants.com/rules/musketeer-chess

The opening position is: rnbekbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR[] w KQBCDFGkqbcdfg - 0 1

With another variable being the side with the chancellor is randomly chosen, so one side starts with the queen and the other the chancellor. Pawns can promote to any piece including knights, bishops, rooks, queens, chancellors, or archbishops.

**Comment by SocialSophia** [comment]
In terms of the versions of pieces I like the looks of these, and they are already sized to the chess board

![Picsart_23-03-03_06-58-39-567](https://user-images.githubusercontent.com/113232613/222648660-98511a91-349b-47ed-a965-1354de20fda9.jpg)
![Picsart_23-03-03_07-16-37-351](https://user-images.githubusercontent.com/113232613/222648698-6f7fb1eb-a4ad-4bb1-8f9b-af322a6ee097.jpg)
![Picsart_23-03-03_07-21-15-829](https://user-images.githubusercontent.com/113232613/222648716-16d240b3-e08f-4c92-a256-01adca010a12.jpg)

**Comment by gbtami** [comment]
I don't really understand how exactly your variant defined, but Musketeer chess and the style piece selection at the beginning of games is not supported by Fairy-Stockfish atm.

**Reply by SocialSophia** [reply]
Thanks for replying, the variant would start with one side with the standard chess 8x8 setup (8 pawns, back rank as: R,N,B,K,Q,B,N,R) and the other side would have the same 8x8 setup, yet the Queen would be replaced with the Chancellor, so R,N,B,K,C,B,N,R. 

The side with the Queen would be randomized each game so you could be white or black with the Queen or Chancellor. An boolean function could address this.

Lastly in terms of the piece selection the "Hawk" (Archbishop) and "Elephant" (Chancellor) already exist, we could update their piece design via 'piece set', the notation can even stay the same for simplicity. 

The starting position for white starting with the Chancellor and black starting with the Queen would be: rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBEKBNR[] w KQBCDFGkqbcdfg - 0 1
And for black starting with the Chancellor and white starting with the Queen would be: rnbhkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR[] w KQBCDFGkqbcdfg - 0 1

If you want me to help with the assets let me know

**Reply by gbtami** [reply]
Thx. If I understand correctly these kind of game is already possible to play on pychess as "custom setup" in S-Chess variant.
![https://i.imgur.com/D24JiQX.png](https://i.imgur.com/D24JiQX.png)

**Reply by SocialSophia** [reply]
Yeah, that's correct if you paste the FEN and remove the EXTRA pieces it's possible, only the queen side wouldn't be random and the piece set is off, my first games (SophiaSocial) show this too. I streamed it and my chat loved the concept. Nice I see you added the new pieces there on the right.

To have this as a default mode would be amazing, so we don't have to copy paste and send link each game. Also Thanks for helping!

**Reply by gbtami** [reply]
You can choose the suggested piece set in Board settings as well (the gear icon on top right, next to the user name).
![https://i.imgur.com/7iGiKBp.png](https://i.imgur.com/7iGiKBp.png)

**Reply by SocialSophia** [reply]
Nice I just seen it! Is the randomized custom setup going to be a new gametype too (you could call it like 'S-Chess Randomized' or 'K-Chess')? 

Could you also add this style to the other pieces in the new piece set? It's from Lichess called something like 'rhoiche', it's angular style is my fav and compliments the set nicely
![Screenshot_20230303_182949_Lichess](https://user-images.githubusercontent.com/113232613/222788655-d394acb2-98bd-499c-baad-af66863f445a.jpg)
![Screenshot_20230303_183033_Lichess](https://user-images.githubusercontent.com/113232613/222788675-91767c7e-db0a-4fb6-a028-4aa6f804d3b6.jpg)

EDIT: After chess analysis, can we do randomized with Chancellor vs Archbishop instead of Chancellor vs Queen. The computer says it's almost perfectly balanced if white starts with the archbishop and flipped is ~1.4, so it'll be more balanced and also it allows BOTH sides to use NEW pieces each game. 
![Screenshot_20230303_190549_Chrome Dev](https://user-images.githubusercontent.com/113232613/222796269-697b0ccb-26ec-4712-8ded-3aedaa4eb2f8.jpg)
![Screenshot_20230303_190619_Chrome Dev](https://user-images.githubusercontent.com/113232613/222796372-35851ef0-c6bd-4f6b-9699-7df52f53e5d4.jpg)
![Screenshot_20230303_190730_Chrome Dev](https://user-images.githubusercontent.com/113232613/222796402-c69b6125-b7b8-4aa6-b21f-a9d7a44100f0.jpg)
![Screenshot_20230303_190815_Chrome Dev](https://user-images.githubusercontent.com/113232613/222796427-47ef7762-5f37-4929-bb5c-a23c1de9d1a6.jpg)

**Comment by gbtami** [comment]
We can add that piece set (it is called riohacha), but unfortunately lichess piece sets doesn't contain archbishop and chancellor pieces at all, so it can be added to Chess variant group (chess, crazyhouse, atomic, placement, duck) only atm.

Telling the truth we don't want to add new variants to the site if they can be played as custom setup games using existing variants, because the number of variants created this way is endless, and if I add one everyone will ask for another one, etc-etc.

Edit: btw it would be an awesome contribution if you can draw riohacha style archbishop and chancellor :)
(Pychess devs use https://inkscape.org to create new piece sets.)

**Reply by SocialSophia** [reply]
Yeah I understand and I'm on the design esp if I can use it for my S-Chess variant. I think we can help many future people with this if we could bring the follow system from Lichess it would allow for easier access. Also if there was a way to save custom positions like presets similar to 'arena tournaments'.

**Reply by gbtami** [reply]
There is a possibility to add custom positions to variants (without the randomization) as you can see in Chess, Capablanca and Shogi.
![https://i.imgur.com/1S8UtdO.png](https://i.imgur.com/1S8UtdO.png)
Maybe we can add Archbishop vs Chancellor setups to S-Chess:
rnbekbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBHKBNR[] w KQkq - 0 1
rnbekbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBHKBNR[] w KQkq - 0 1

**Reply by gbtami** [reply]
Yea, lichess has enormous amount of useful features we don't have. We try to add some of them gradually, but our dev resources are rather limited. So any contribution is welcome! :)

**Comment by SocialSophia** [comment]
Tami I respect all contributions now, this took hours to make! It's an nearly perfect model of an archbishop in the riohacha style, 4K quality. The shading was a challenge, feel free to edit if it's not perfect. 
![dce460708f814db9b4fe93af473c5bba](https://user-images.githubusercontent.com/113232613/222858363-706cbdb7-19e5-46a9-afa4-884edb026ef4.png)
And yeah that would be amazing if we had a Chancellor vs Archbishop in settings! I'm working on the Chancellor currently

**Reply by gbtami** [reply]
Wait! Riocacha pieces are all vector graphics (.svg) not pixel graphics (.png) on https://github.com/lichess-org/lila/tree/master/public/piece/riohacha
This is important, because vector graphics images scale well to any screen size.

**Reply by SocialSophia** [reply]
I'm on it
